### ISSUE TITLE
failure to clone repo -- need to somehow urlencode username

### ISSUE THREAD

**Issue by dlethin** [ashwanthkumar/gocd-build-github-pull-requests#84]
I see the following error in the log file:

```
2017-01-18 20:19:07,622  INFO [92@MessageListener for MaterialUpdateListener] GitHubPRBuildPlugin:52 - Flyweight: /var/lib/go-server/pipelines/flyweight/3d06d640-9ed8-4fce-a0d1-97fa1e939bd9
2017-01-18 20:19:07,640  WARN [92@MessageListener for MaterialUpdateListener] GitHubPRBuildPlugin:67 - get latest revision: 
java.lang.RuntimeException: Exception Occurred: [git, clone, --branch=master, https://me+github@mydomain.com:mypassword@github.com/path/to/my/repo, /var/lib/go-server/pipelines/flyweight/3d06d640-9ed8-4fce-a0d1-97fa1e939bd9] - null
        at com.tw.go.plugin.cmd.Console.runOrBomb(Console.java:33)
        at com.tw.go.plugin.git.GitCmdHelper.runAndGetOutput(GitCmdHelper.java:397)
        at com.tw.go.plugin.git.GitCmdHelper.cloneRepository(GitCmdHelper.java:55)
        at com.tw.go.plugin.GitHelper.cloneOrFetch(GitHelper.java:38)
        at in.ashwanthkumar.gocd.github.GitHubPRBuildPlugin.handleGetLatestRevision(GitHubPRBuildPlugin.java:215)
        at in.ashwanthkumar.gocd.github.GitHubPRBuildPlugin.handle(GitHubPRBuildPlugin.java:119)
        at com.thoughtworks.go.plugin.infra.DefaultPluginManager$1.execute(DefaultPluginManager.java:172)
        at com.thoughtworks.go.plugin.infra.DefaultPluginManager$1.execute(DefaultPluginManager.java:167)
        at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.executeActionOnTheService(FelixGoPluginOSGiFramework.java:315)
        at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.doOn(FelixGoPluginOSGiFramework.java:245)
        at com.thoughtworks.go.plugin.infra.DefaultPluginManager.submitTo(DefaultPluginManager.java:167)
        at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:32)
        at com.thoughtworks.go.plugin.access.scm.SCMExtension.getLatestRevision(SCMExtension.java:109)
        at com.thoughtworks.go.server.service.materials.PluggableSCMMaterialPoller.latestModification(PluggableSCMMaterialPoller.java:62)
        at com.thoughtworks.go.server.service.materials.PluggableSCMMaterialPoller.latestModification(PluggableSCMMaterialPoller.java:47)
        at com.thoughtworks.go.server.service.MaterialService.latestModification(MaterialService.java:105)
        at com.thoughtworks.go.server.materials.LegacyMaterialChecker.findLatestModification(LegacyMaterialChecker.java:49)
        at com.thoughtworks.go.server.materials.ScmMaterialUpdater.insertLatestOrNewModifications(ScmMaterialUpdater.java:53)
        at com.thoughtworks.go.server.materials.ScmMaterialUpdater.addNewMaterialWithModifications(ScmMaterialUpdater.java:73)
        at com.thoughtworks.go.server.materials.PluggableSCMMaterialUpdater.addNewMaterialWithModifications(PluggableSCMMaterialUpdater.java:66)
        at com.thoughtworks.go.server.materials.MaterialDatabaseUpdater.addNewMaterialWithModifications(MaterialDatabaseUpdater.java:177)
        at com.thoughtworks.go.server.materials.MaterialDatabaseUpdater.initializeMaterialWithLatestRevision(MaterialDatabaseUpdater.java:134)
        at com.thoughtworks.go.server.materials.MaterialDatabaseUpdater.access$000(MaterialDatabaseUpdater.java:47)
        at com.thoughtworks.go.server.materials.MaterialDatabaseUpdater$1.doInTransaction(MaterialDatabaseUpdater.java:95)
        at com.thoughtworks.go.server.transaction.TransactionCallback.doWithExceptionHandling(TransactionCallback.java:24)
        at com.thoughtworks.go.server.transaction.TransactionTemplate$3.doInTransaction(TransactionTemplate.java:53)
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:130)
        at com.thoughtworks.go.server.transaction.TransactionTemplate.executeWithExceptionHandling(TransactionTemplate.java:49)
        at com.thoughtworks.go.server.materials.MaterialDatabaseUpdater.updateMaterial(MaterialDatabaseUpdater.java:92)
        at com.thoughtworks.go.server.materials.MaterialUpdateListener.onMessage(MaterialUpdateListener.java:50)
        at com.thoughtworks.go.server.materials.MaterialUpdateListener.onMessage(MaterialUpdateListener.java:30)
        at com.thoughtworks.go.server.messaging.activemq.JMSMessageListenerAdapter.runImpl(JMSMessageListenerAdapter.java:65)
        at com.thoughtworks.go.server.messaging.activemq.JMSMessageListenerAdapter.run(JMSMessageListenerAdapter.java:50)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 128 (Exit value: 128)
        at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
        at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
        at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:153)
        at com.tw.go.plugin.cmd.Console.runOrBomb(Console.java:25)
        ... 33 more

```

I get the same error message when I run that git clone command directly in a terminal.  However, if I urlencode my username, the commandline run works...  so I though I could then store my urlencoded username in the ~/.github file.  However, I am still getting the same error.

Is the information in ~/.github cached somewhere that I need to clear out.  Or do I need to escape the username somehow for this to work correctly?

I'm using github-pr-poller-1.3.0-RC2.jar   and Go-CD 16.7.0

Thanks so much.

**Comment by ashwanthkumar** [comment]
@dlethin The information in `~/.github` is used only for making Github API requests not for cloning. Can you try changing the url in the Github Material of the pipeline to the encoded version and try?

**Comment by dlethin** [comment]
Thanks, @ashwanthkumar.  That was the magic I was looking for.  I didn't even think of that.  I think what confused me was that after initially setting username/password in the github materials, I removed it, thinking it was duplicated from the information in ~/.github, and then even though I removed it (and restarted go-server), I was still seeing that error.  It seems perhaps go-server must be caching those credentials somewhere?

Anyway, I got past my initial error and hopefully the rest of integrating with this plugin will be smooth.  Thanks again.
